front/lib/models/assistant/actions/retrieval.ts

```diff
@@ -0,0 +1,214 @@
+import {
+  DataTypes,
+  ForeignKey,
+  InferAttributes,
+  InferCreationAttributes,
+  Model,
+} from "sequelize";
+
+import { front_sequelize } from "@app/lib/databases";
+import { AssistantAgentConfiguration } from "@app/lib/models/assistant/agent";
+import { DataSource } from "@app/lib/models/data_source";
+import { TimeframeUnit } from "@app/types/assistant/actions/retrieval";
+
+/**
+ * Action Retrieval configuration
+ */
+export class AssistantAgentRetrievalConfiguration extends Model<
+  InferAttributes<AssistantAgentRetrievalConfiguration>,
+  InferCreationAttributes<AssistantAgentRetrievalConfiguration>
+> {
+  declare id: number;
+
+  declare query: "auto" | "none" | "templated";
+  declare queryTemplate: string | null;
+  declare relativeTimeFrame: "auto" | "none" | "custom";
+  declare relativeTimeFrameDuration: number | null;
+  declare relativeTimeFrameUnit: TimeframeUnit | null;
+  declare topK: number;
+
+  declare agentId: ForeignKey<AssistantAgentConfiguration["id"]>;
+}
+AssistantAgentRetrievalConfiguration.init(
```

Comment:
On the type we have `dataSources: "all" | DataSourceConfiguration[];` 
I wondered if I should add a boolean `allowAllDatasources` but I figured that it could be deduced depending on wether there's attached AssistantAgentDatasourceConfiguration or not. WDYT?

Reply:
Fine as is :+1: cc @fontanierh @dust-tt/engineering-team 

As this will likely push us to use the same kind of pattern in other places